### ISSUE TITLE
Should we keep UAttributesBuilder?

### ISSUE THREAD

**Issue by evshary** [eclipse-uprotocol/up-rust#55]
I love the PR #46, which makes the code cleaner.
However, I found that we still need to create the UAttributes by ourselves inside `invoke_method` while sending the RPC request.

```rust
async fn invoke_method(
    &self,
    method: UUri,
    request: UPayload,
    options: CallOptions,
) -> RpcClientResult;
```

Now I use `UMessageBuilder` to create the UMessage and extract the UAttributes.
https://github.com/eclipse-uprotocol/up-client-zenoh-rust/pull/5/files#diff-5cdcc3261dd77eadb37bcd60c5f85affcc9306454ae7bb8128e6d096ae51a631R61
Maybe we can still keep UAttributesBuilder here?

@sophokles73  Do you have any suggestions?

**Comment by sophokles73** [comment]
My understanding is that you need the UAttributes because you want to put their protobuf encoding into the zenoh message attachment, right? Maybe we can add a
```rust
pub fn request(message_id: UUID, method: UUri, reply_to_address: UUri, options: CallOptions) -> UAttributes
```
factory function to `UAttributes` for this specific purpose? I do not believe that we need a fully flexible builder at this level.

**Comment by evshary** [comment]
Correct. I think this is a proper solution.

**Comment by PLeVasseur** [comment]
I think the proposed function should also take a request ID as well, like this:
```rust
pub fn request(message_id: UUID, method: UUri, request_id: UUID, reply_to_address: UUri, options: CallOptions) -> UAttributes
```
because the UAttributes formed here should be "signed" with a UUID built using a UUIDBuilder from the current uE. So there would be a call to e.g. `uuid_builder.build()` first to get the `request_id` argument to pass in.

**Comment by sophokles73** [comment]
@PLeVasseur based on the current `UMessageBuilder` code you are right. However, I am working on several changes/improvements already. One of them being the insight, that request messages do not carry a separate request_id but instead the message_id is used as the correlation ID when a response wants to reply.

**Comment by PLeVasseur** [comment]
OK. I'll hang tight and wait to check out the PR :slightly_smiling_face:

**Comment by sophokles73** [comment]
closed via #64
